pkg/kv/kvserver/scheduler.go

```diff
@@ -453,6 +453,12 @@ func (ss *raftSchedulerShard) worker(
 			//   and the worker does not go back to sleep between the current
 			//   iteration and the next iteration, so no change to num_signals
 			//   is needed.
+			//
+			// NB: we overwrite state.begin unconditionally since the next processing
+			// can not possibly happen before the current processing is done (i.e.
+			// now). We do not want the scheduler latency to pick up the time spent
+			// handling this replica.
+			state.begin = crtime.NowMono()
```

Comment:
Is this a no-op? Should we rather update `ss.state[id].begin`?

Reply:
No, this wasn't it - I had trouble even elevating the baseline latency on master. The bug here was real and is now fixed, though, and I added a unit test.